prometheus-ksonnet/lib/alertmanager.libsonnet

```diff
@@ -121,14 +121,24 @@
     ], self.alertmanager_pvc) +
     statefulset.mixin.spec.withServiceName('alertmanager') +
     statefulset.mixin.spec.template.metadata.withAnnotations({ 'prometheus.io.path': '%smetrics' % $._config.alertmanager_path }) +
-    statefulset.mixin.spec.template.spec.securityContext.withRunAsUser(0) +
-    statefulset.mixin.spec.template.spec.securityContext.withFsGroup(0) +
     $.util.configVolumeMount('alertmanager-config', '/etc/alertmanager/config') +
     $.util.podPriority('critical')
   else {},
 
+  local service = $.core.v1.service,
+  local servicePort = service.mixin.spec.portsType,
+
   // Do not create service in clusters without any alertmanagers.
-  alertmanager_service: if replicas > 0 then
-    $.util.serviceFor($.alertmanager_statefulset)
-  else {},
+  alertmanager_service:
+    if replicas == 0
+    then {}
+    else
+      $.util.serviceFor($.alertmanager_statefulset) +
+      service.mixin.spec.withPortsMixin([
```

Comment:
You're using `withPortsMixin` here – wouldn't that add another port to the service, along the one inferred from the deployment? Is this intentional?

Reply:
Yes, I wanted to leave all the existing ports there just in case they are being used by name.